pkg/instance/repository.go

```diff
@@ -128,3 +142,62 @@ func (r repository) FindByGroupNames(names []string) ([]*model.Instance, error)
 
 	return instances, err
 }
+
+func (r repository) encryptParameters(instance *model.Instance) error {
+	for i, parameter := range instance.RequiredParameters {
+		value, err := encryptText(r.config.InstanceParameterEncryptionKey, parameter.Value)
+		if err != nil {
+			return err
+		}
+		instance.RequiredParameters[i].Value = value
+	}
+
+	for i, parameter := range instance.OptionalParameters {
+		value, err := encryptText(r.config.InstanceParameterEncryptionKey, parameter.Value)
+		if err != nil {
+			return err
+		}
+		instance.OptionalParameters[i].Value = value
+	}
+
+	return nil
+}
+
+func (r repository) decryptParameters(instance *model.Instance) error {
+	for i, parameter := range instance.RequiredParameters {
+		value, err := decryptText(r.config.InstanceParameterEncryptionKey, parameter.Value)
+		if err != nil {
+			return err
+		}
+		instance.RequiredParameters[i].Value = value
+	}
+
+	for i, parameter := range instance.OptionalParameters {
+		value, err := decryptText(r.config.InstanceParameterEncryptionKey, parameter.Value)
+		if err != nil {
+			return err
+		}
+		instance.OptionalParameters[i].Value = value
+	}
+
+	return nil
+}
+
+// TODO: Rename PopulateRelations? Or something else?
+func enrichParameters(instance *model.Instance) {
+	requiredParameters := instance.RequiredParameters
+	if len(requiredParameters) > 0 {
```

Comment:
I think 😋 ranging over `instance.RequiredParameters` should not require `len` check https://groups.google.com/g/golang-nuts/c/gVgVLQU1FFE

try https://go.dev/play/p/XcbTTnfzLwM